app/Libraries/Generators/MediaGenerator.php

```
@@ -46,24 +45,32 @@ public static function generateAndStorePhotoFiles($projectRef, $filename, $forma
-            // Draw the star polygon
-            $img->polygon($points, function ($draw) {
-                $draw->background('#' . dechex(rand(0x000000, 0xFFFFFF)));
+            // Generate random color for the star
+            $starColor = '#' . str_pad(dechex(rand(0x000000, 0xFFFFFF)), 6, '0', STR_PAD_LEFT);
+
+            // Draw the star polygon using closure
+            $img->drawPolygon(function ($polygon) use ($points, $starColor) {
+                // Convert flat array to point pairs and add to polygon
+                for ($k = 0; $k < count($points); $k += 2) {
+                    $polygon->point($points[$k], $points[$k + 1]);
+                }
```

PHPMD (2.15.0)

Avoid using ec5\Libraries\Generators\count() function in for loops. (Design Rules)

(CountInLoopExpression)